File: CarRentalApi/CarRentalApi/Database/Model/Reservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace CarRentalApi.Database.Model
{
    public class Reservation
    {
        public int ReservationNumber { get; set; }


        public int PickUpLocationId { get; set; }
        public Location PickUpLocation { get; set; }

        public int ReturnLocationId { get; set; }
        public Location ReturnLocation { get; set; }

        public DateTime PickUpDate { get; set; }
        public DateTime ReturnDate { get; set; }

        public int CarId { get; set; }
        public Car Car { get; set; }

        public string Surname { get; set; }
        public int Age { get; set; }

    }
}

File: car-rental-app/src/app/models/reservation.ts
import { Location } from './location';
import { Car } from './car';

export class Reservation {
  reservationNumber: number;
  pickUpLocationId: number;
  returnLocationId: number;
  pickUpDate: Date;
  returnDate: Date;
  carId: number;
  surname: string;
  age: number;
}
export function InitReservation() : Reservation{
  return {
    reservationNumber : 0,
    pickUpLocationId : 0,
    returnLocationId : 0,
    pickUpDate : new Date(),
    returnDate: new  Date(),
    carId: 0,
    surname: '',
    age: 0
  };
}
export class ReservationDetail{
  reservationNumber: number;
  pickUpLocation: Location;
  returnLocation: Location;
  pickUpDate: Date;
  returnDate: Date;
  car: Car;
  surname: string;
  age: number;
  totalPrice: number;
}

export function detailToReservation(detail: ReservationDetail) : Reservation{
  return {
    reservationNumber: detail.reservationNumber,
    pickUpLocationId: detail.pickUpLocation.id,
    returnLocationId: detail.returnLocation.id,
    pickUpDate: detail.pickUpDate,
    returnDate: detail.returnDate,
    carId: detail.car.id,
    surname: detail.surname,
    age: detail.age
  };
}